docs/EUDB-compliance.md

```diff
@@ -0,0 +1,65 @@
+# EUDB Guidance for 1DS C++ SDK
+
+In order to satisfy the Microsoft commitment to ensuring Privacy and Compliance, specifically
+EUDB compliance with respect to EU 'Schrems II' decision, it is imperative for certain
+commercial products to perform EUDB URL upload determination during application launch.
+
+1DS Collector service accepts data from all One Observability client SDKs. By default, traffic
+simply flows to whichever region can best handle the traffic. This approach works well for
+system required metadata. However some client scenarios require that data is sent to a specific
+geographic location only.
+
+1DS C++ SDK supports ability to specify / adjust the upload URL at runtime.
+
+Two approaches could be applied to implement EUDB-compliant data upload.
+
+## Option 1: Create two instances of 1DS C++ SDK - one for US collector, another for EU collector
+
+See [Multiple Log Managers Example](https://github.com/microsoft/cpp_client_telemetry/tree/main/examples/cpp/SampleCppLogManagers)
+that illustrates how to create multiple instances, each acting as a separate vertical pillar with
+their own data collection URL. Two instances `LogManagerUS` and `LogManagerEU` may be configured
+each with their own data collection URL, for example:
+
+- For US customers: `https://us-mobile.events.data.microsoft.com/OneCollector/1.0/`
+- For EU customers: `https://eu-mobile.events.data.microsoft.com/OneCollector/1.0/`
+
+Depending on data requirements and outcome of dynamic EUDB determination, i.e. organization /
+M365 Commercial Tenant is located in EU, the app decides to use `LogManagerEU` instance for
+telemetry. Default `LogManager` instance can still be used for region-agnostic "global"
+collection of required system diagnostics data. Remember to use the proper compliant instance
+depending on event type.
+
+## Option 2: Autodetect the corresponding data collection URL on app start
```

Comment:
Is this logic applicable during application start, or else can something below done while event transmission is happening:

 `url_change_detected` -> `PauseTransmission` -> `update Logconfiguration::url` -> `ResumeTransmission` 

Or we can't recommend this as there is the possibility of events getting leaked to incorrect endpoint.

Reply:
I can elaborate on this further in the document if you like

- `initialize` with no URL - functionally equivalent to "paused" state.
- `pause` - to commit changes to uploader configuration in thread-safe manner.
- `detect` the url ONCE, after the URL determination is complete. It's once only, typically shortly after app start, and/or immediately after the app comes online, and once user is logged in. Set the URL in configuration.
- `resume` - for new uploader configuration to take effect.

There's no intention to run a periodic URL detection. It's done once, typically after user is logged in, OR when the url for a given application (seat) is known based on some other configuration cache.

Reply:
Thanks, @magolov, should we explicitly mention that the URL should be changed only one during startup, and NOT afterward?

Reply:
Sounds good. I'll add an explanation for this in the markdown.

Reply:
Added an additional explanation that the URL is determined and set once only, during app start.